pydantic/_internal/_validators.py

```diff
@@ -316,6 +322,15 @@ def min_length_validator(x: Any, min_length: Any) -> Any:
             )
         return x
     except TypeError:
+        if isinstance(x, pathlib.PurePath):
+            try:
+                if not (len(str(x)) >= min_length):
+                    raise PydanticKnownError(
+                        'too_short', {'field_type': 'Value', 'min_length': min_length, 'actual_length': len(str(x))}
+                    )
+                return x
+            except TypeError:
+                pass
```

Comment:
IMO it would be more correct to convert this to `bytes()` instead of `str()`

Reply:
https://docs.python.org/3/library/pathlib.html

> Calling [bytes](https://docs.python.org/3/library/stdtypes.html#bytes) is only recommended under Unix. Under Windows, the unicode form is the canonical representation of filesystem paths.

Also I think people will mean "the number of characters" not "the number of bytes".

Idk what the right thing to do is 😢.